### ISSUE TITLE
Canu 1.4.1 failed with bus error (Failed with exit code 135.)

### ISSUE THREAD

**Issue by dbayles** [marbl/canu#394]
I'm running CANU on a cluster, but restricting execution to a single high memory node.  CANU is being launched with: "canu -p 2kbpd -d plasmidome-auto genomeSize=1m -corOutCoverage=1000 -pacbio-raw 2kb_plasmidome.fastq.gz useGrid=false > canu_terminal.txt 2>&1".

Late in the process at the ovStoreBuild step, the software is failing with the following message.
```
-- Finished stage 'overlapErrorAdjustmentCheck', reset canuIteration.
----------------------------------------
-- Starting command on Sat Mar  4 08:26:37 2017 with 1312781.116 GB free disk space

    cd unitigging/3-overlapErrorAdjustment
    /software/apps/canu/gcc/64/1.4.1/ovStoreBuild \
      -G ../2kbpd.gkpStore \
      -O ../2kbpd.ovlStore \
      -evalues \
      -L ./oea.files \
    > ./oea.apply.err 2>&1
sh: line 5: 87149 Bus error               /software/apps/canu/gcc/64/1.4.1/ovStoreBuild -G ../2kbpd.gkpStore -O ../2kbpd.ovlStore -evalues -L ./oea.files > ./oea.apply.err 2>&1

-- Finished on Sat Mar  4 08:26:47 2017 (10 seconds) with 1312780.034 GB free disk space
----------------------------------------
ERROR:
ERROR:  Failed with exit code 135.  (rc=34560)
ERROR:
================================================================================
Don't panic, but a mostly harmless error occurred and canu failed.
```

**Comment by brianwalenz** [comment]
Which `canu -version` is this?
What is in oea.apply.err?  Empty file?

It might have been fixed in the latest code, but the pipeline might not play nicely with outputs from an older version.  Running that same command but using the latest code should work, then once it finishes, go back to the original version to finish the assembly.

If you can share those files, I can try to reproduce here.

**Comment by dbayles** [comment]
Canu 1.4.1 should be the version I'm currently using.  Running the version command yields a perplexing result:
$ canu --version
Canu v0.0 (+0 commits) r0 unknown-hash-tag-no-repository-available.

I've attached the oea.apply.err, and note that the bus error is immediately upstream of what gets written to the oea.apply.err file.  I modified the extension to be *.txt to facilitate uploading the file.
[oea.apply.err.txt](https://github.com/marbl/canu/files/822386/oea.apply.err.txt)

**Comment by dbayles** [comment]
I just checked several other versions of CANU that are available on the cluster that I'm using, and regardless of the CANU module I load, I get exactly the same version information that I included in the previous comment.

**Comment by HenrivdGeest** [comment]
Bus errors can also be caused by running a binary from a NFS share, Try to put CANU on another storage system, or even locally.

**Comment by brianwalenz** [comment]
The lack of a version is caused by building from source that isn't a git checkout.  The github generated source releases ("Source code (zip)" and "Source code (tar.gz)" do this.  What's confusing me is that we don't have a 1.4.1 release.  I'm guessing you applied some patches to 1.4.

I suspect this is caused by line 558 in ovStore.C ("delete [] evalues;").  Remove that line.

If you've got the 'evalues' file in 2kbpd.ovlStore, you can (probably) just restart and canu will happily continue.

**Comment by dbayles** [comment]
There are files of significant size in the correction/2kbpd.ovlStore/, trimming//2kbpd.ovlStore/, and unitigging/2kbpd.ovlStore/ directories.

I don't know the source of the versioning information since one of the system admins downloaded and compiled the software.  They may have created the versioning to simply differentiate from the v. 1.4 install that was done previously.  I have access to a build node and can try your suggestion of editing the ovStore.C file and then compiling.

I did try running CANU using the v. 1.4 precompiled binary, and the exact same error at the same step occurred.  There may be an issued with some of the reads having a lot of repetitive sequence too.

In response to a previous comment, the filesystem is Lustre.

**Comment by brianwalenz** [comment]
On second glance, line 558 isn't the problem, so we're back to the start.  If the restart doesn't work, let me know.

**Comment by dbayles** [comment]
I've previously tried restarting by reissuing the original command and it errors out at the same point with the same message.

**Comment by brianwalenz** [comment]
That was helpful.  I think it's an incompatibility with Lustre.  I'm doing something bizarre to update values in this 'evalues' file.

Try moving unitigging/2kbpd.ovlStore/evalues file to a non-Lustre directory, /tmp would work, and creating a symlink to it from the ovlStore directory.  Run the ovStoreBuild command - it's lightweight and can probably be run from a head node if needed.  All it is doing is copying data from the input files (files listed in the -L option) to that 'evalues' file.  If it works, just move the file back and restart.

The fix will take a small bit of engineering.

**Comment by dbayles** [comment]
O.k.  I'm understanding your earlier comment a bit differently now.  Looks like the 2kbpd.ovlStore/evalues file was not created when the process quit initially (i.e. the file doesn't exist).  While there are files of size in that directory, there is no named "evalues" files.  Here's a listing of what's in 2kbpd.ovlStore/:
$ ls unitigging/2kbpd.ovlStore/
0001  0002  0003  0004  0005  0006  0007  0008  0009  evalueLen  index  info

**Comment by skoren** [comment]
Has this been resolved? We've changed how the evalues are processed so it should fix the error if you update to the latest (1.5) release. However, you cannot resume an existing run in the middle with the new code that was made with the old code. You can run with the corrected reads (specifying pacbio-corrected or nanopore-corrected) or from the trimmed reads (specifying -assemble) instead. If you encounter issues in the new version, open a new issue.

For future users who encounter bus errors, these can also be caused by job begin killed by the grid for exceeding memory limits. In that case, increasing the memory request from the default for the failing step (```canu -options |grep Memory``` to get a list of options) would help.
